github/GithubRetry.py

```diff
@@ -112,7 +112,19 @@ def increment(
                         content = self.get_content(response, url)
                         content = json.loads(content)
                         message = content.get("message")
+                    except Exception as e:
+                        # we want to fall back to the actual github exception (probably a rate limit error)
+                        # but provide some context why we could not deal with it without another exception
+                        try:
+                            raise RuntimeError(
+                                "Failed to inspect response message"
+                            ) from e
+                        except RuntimeError as e:
```

Comment:
why make it so complicated here

Reply:
Is this the reason you suggest this ?

https://github.com/PyGithub/PyGithub/pull/2591#issuecomment-1619670289

Reply:
The response from Github causes the `GithubException`, which we want to prevent by retrying, but the logic to determine the right backoff raised some exception itself, which should never happen. So I want to preserve the inner exception with stacktrace to better understand and debug the root cause, amend it with some meaningful error message (`RuntimeError`), but still provide the usual `GithubException` based on the response from Github.

The chain is more appropriate here than adding an error message to the outer `GithubException`.

I could not find a nicer syntax to do the chaining, something like `raise GithubException() from RuntimeError() from e`.

Reply:
maybe exception group? https://peps.python.org/pep-0654/

Reply:
I think we can add a extra message field in GithubException and use `"Failed to inspect response message"` as exception message?

Reply:
Yes, we should add the message field, but I would not use it here, because that error message is not coming from / caused by Github, but from PyGithub library. Hence the cause chain.